lib/cloudformation.py

```diff
@@ -32,6 +33,19 @@
 from .migrations import MigrationManager
 from .exceptions import BossManageError, BossManageCanceled
 
+"""
+Input for CloudFormationConfiguration.add_app_loadbalancer().
+
+Defines configuration for a listener.
+
+Fields:
+    in_port (str): Port to listen on.
+    to_port (str): Port to send traffic to.
+    protocol (str): HTTP or HTTPS.
+    cert (None|str): ARN of SSL certificate.
+"""
+AppLoadBalancerListenerCfg = namedtuple('AppLoadBalancerListenerCfg', ['in_port', 'to_port', 'protocol', 'cert'])
```

Comment:
While more descriptive, it doesn't really match with the rest of how data is passed to the previous load balancer configuration or how the other `add_` methods pass data. It also splits the documentation between the top of the file and the `add_app_loadbalancer()` method.

Reply:
I converted to `namedtuple` in the last commit, so I can easily revert it if you rather keep all the `add_` methods consistent. I switched to this because I thought it'd be more maintainable and easier to follow what the code is doing. I also thought I wasn't using the `to_port`. That was the original reason for converting. I thought it'd be safer than making sure I decremented the right indices. Then when doing a final search for `listener[`, I realized that I was still using the `to_port`. 😛